turbinia/lib/docker_manager.py

```diff
@@ -0,0 +1,222 @@
+    try:
+      image_verif = self.client.images.get(image_id)
+    except docker.errors.ImageNotFound as exception:
+      message = 'The Docker image {0!s} could not be found: {1!s}'
+      raise TurbiniaException(message.format(image_id, exception))
+    return image_verif
+
+  def list_images(self, filters=None):
+    """Lists all available Docker images.
+
+    Args:
+     filters(str): If provided, will return a subset of the Images data.
+        Allowed values are 'short_id' and 'id'.
+
+    Returns:
+      images(list): A list of available Docker images.
+
+    Raises:
+      TurbiniaException: If an error occured retrieving the images.
+    """
+    accepted_vars = ['short_id', 'id']
+    try:
+      images = self.client.images.list()
+      if filters in accepted_vars:
+        # short_id and id will always start with sha256:
+        images = [
+            getattr(img, filters).replace('sha256:', '') for img in images
+        ]
+    except docker.errors.APIError as exception:
+      raise TurbiniaException(
+          'An error occured retrieving the images: {0!s}'.format(exception))
+    return images
+
+
+class ContainerManager(DockerManager):
+  """Class representing Docker containers.
+
+  Attributes:
+    image_id(str): Docker image id.
+  """
+
+  def __init__(self, image_id):
+    """Initialize the ContainerManager object.
+
+    Args:
+      image_id(str): The image id to create a container from.
+    """
+    super(ContainerManager, self).__init__()
+    self.image_id = self.verify_image(image_id)
+
+  def _create_mount_points(self, mount_paths):
+    """Creates file and device mounting arguments.
+
+    The arguments will be passed into the container and all device blocks
+    will be mounted as ro while file paths would be mounted as rw.
+
+    Attributes:
+      mount_paths(list): The paths on the host system to be mounted.
+
+    Returns:
+      device_paths(list): device blocks that will be mounted.
+      file_paths(dict): file paths that will be mounted.
+    """
+    device_paths = []
+    file_paths = {}
+
+    for mpath in mount_paths:
+      if mpath not in file_paths.keys() or mpath not in device_paths:
+        if IsBlockDevice(mpath):
+          formatted_path = '{0:s}:{0:s}:{1:s}'.format(mpath, 'r')
+          device_paths.append(formatted_path)
+        else:
+          file_paths[mpath] = {'bind': mpath, 'mode': 'rw'}
+    return device_paths, file_paths
+
+  def execute_container(self, cmd, mount_paths=None, **kwargs):
+    """Executes a Docker container.
+
+    A new Docker container will be created from the image id,
+    executed, and then removed.
+
+    Attributes:
+      cmd(str|list): command to be executed.
+      mount_paths(list): A list of paths to mount to the container.
+      **kwargs: Any additional keywords to pass to the container.
+
+    Returns:
+      stdout(str): stdout of the container.
+      stderr(str): stderr of the container.
+      ret(str): the return code of process run.
```

Comment:
Is this actually a string and not an int?

Reply:
Thanks.  Sorry, I didn't catch this before, but maybe (small nit): s/of process run/of the process run/.

Reply:
(Ping for visibility since I think this got hidden)

Reply:
fixed